shared_utils.py

```diff
@@ -117,7 +118,15 @@ def check_styles(soup: bs4.BeautifulSoup) -> None:
         warn('style_no_intro')
     if not get_elem_containing_text(soup, 'h1', 'references'):
         warn('style_no_refs')
-
+    # Check every numbered reference appears in the text in square brackets
+    dom = etree.HTML(str(soup))
+    ref_li = dom.xpath("//h1[contains(text(), 'REFERENCES')]/following-sibling::ol/li | //h1[.//*[contains(text(), 'REFERENCES')]]/following-sibling::ol/li") 
+    ref_in_ref = set( range(1,len(ref_li)+1)  ) 
+    ref_in_text_raw = re.findall('\[([\d, ]+)\]', soup.get_text( strip=True) )
+    ref_in_text = set( [ int(i) for i in re.split(r'\D+',','.join(ref_in_text_raw)) ] )
```

Comment:
This crashes if there are no references in the document, but I am not worried about that at all since it seems like not a situation that would ever occur! And the author would probably get a message about no "References" section anyway.

Reply:
I can also wrap in try/catch - shall I do that?

Reply:
Excellent idea. I thought about it more and it *might* indeed be a situation that occurs if the author uses the wrong in-text citation style, such as APA (seems plausible). So failing gracefully would be good.

Reply:
I put in extra warnings in case there are no in-text citations or no references found. I think this avoids the need for try/catch